aicsimageio/writers/rgb_writer.py

```diff
@@ -0,0 +1,104 @@
+
+
+class RGBWriter(Writer):
+    """
+    A writer for image data where the channel dimension act's as color. (RGB / RGBA)
+    Primarily directed at formats: "png", "jpg", etc.
+
+    This is primarily a passthrough to imageio.imwrite.
+    """
+
+    @staticmethod
+    def save(
+        data: Union[da.Array, np.ndarray],
+        filepath: Union[str, Path],
+        dim_order: str = None,
+        **kwargs,
+    ):
+        """
+        Write a data array to a file.
+
+        Parameters
+        ----------
+        data: Union[da.Array, np.ndarray]
+            The array of data to store. Data must have either two or three dimensions.
+        filepath: Union[str, Path]
+            The path to save the data at.
+        dim_order: str
+            The dimension order of the provided data.
+            Default: None. Based off the number of dimensions, will assume
+            the dimensions similar to how aicsimageio.readers.DefaultReader reads in
+            data. That is, two dimensions: YX and three dimensions: YXC.
+
+        Examples
+        --------
+        Data is the correct shape and dimension order
+
+        >>> image = dask.array.random.random((100, 100, 4))
+        ... RGBWriter.save(image, "file.png")
+
+        Data provided with current dimension order
+
+        >>> image = numpy.random.rand(3, 1024, 2048)
+        ... RGBWriter.save(image, "file.png", "CYX")
+        """
+        # Check filepath
+        # Do not fully resolve to absolute because Mac is bad with mounted drives
+        filepath = Path(filepath).expanduser()
+
+        # Assumption: if provided a dask array to save, it can fit into memory
+        if isinstance(data, da.core.Array):
+            data = data.compute()
+
+        # Check num dimensions
+        if len(data.shape) not in [2, 3]:
+            raise InconsistentShapeError(
+                f"RGBWriter requires that data must have either 2 or 3 dimensions. "
+                f"Provided data with {len(data.shape)} dimensions. ({data.shape})"
+            )
+
+        # Assume dim order if not provided
+        if dim_order is None:
+            if len(data.shape) == 2:
+                dim_order = GREYSCALE_ORDER
+            if len(data.shape) == 3:
+                dim_order = RGB_ORDER
+
+        # Uppercase dim order
+        dim_order = dim_order.upper()
+
+        # Check dimensions provided in the dim order string are all C, Y, or X
+        if any([dim not in ["C", "Y", "X"] for dim in dim_order]):
+            raise InvalidDimensionOrderingError(
+                f"The dim_order parameter only accepts dimensions: 'C', 'Y', or 'X' "
+                f"Provided dim_order string: '{dim_order}'."
+            )
+
+        # Transpose dimensions if dim_order not ready for imageio
+        if len(data.shape) == 2 and dim_order != GREYSCALE_ORDER:
+            data = reshape_data(data, given_dims=dim_order, return_dims=GREYSCALE_ORDER)
```

Comment:
Thought here... what if the first check in `reshape_data` was
```python
if given_dims == return_dims:
    return data
```
Then the `*_ORDER` dims could actually be a map of data shape length to dim order:
```
DIM_ORDERS = {
    2: "YX",
    3: "YXC",
}
```
This would simplify the logic on L81 and could simplify the logic here too:
```
# This would always be run, but where dim_order == DIMS_ORDER[len(data.shape)] it would be a noop
data = reshape_data(data, given_dims=dim_order, return_dims=DIMS_ORDER[len(data.shape)])
```

Just a thought, this requires some other changes so it's totally up to your discretion :slightly_smiling_face:

Reply:
> Thought here... what if the first check in `reshape_data` was
> 
> ```python
> if given_dims == return_dims:
>     return data
> ```

or return a copy, I don't think reshape_data is supposed to return the original ref, is it?

Reply:
I like all of these suggestions. Will add to the PR.

Reply:
> I don't think reshape_data is supposed to return the original ref, is it?

And no, `reshape_data` always returns a new array.

Reply:
which means it would be an unnecessary copy in the case of same dims.

Reply:
Ahhhh I see your point. In which case I think I will at least make the `DIM_ORDERS` changes but probably not add the initial check to `reshape_data`.